src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -771,6 +783,211 @@ bool EnzymeTreeItem::operator<(const QTreeWidgetItem& other) const {
+         reverseTooltipElements = generateTooltipElements(reverseNOut, reverseNIn, false, reverseNShift.first,
+                                                         (forwardNOut != 0 && forwardNShift.first == Ns::Left),
+                                                         (forwardNIn != 0 && forwardNShift.first == Ns::Left));
+    } else {
+        if (forwardNShift.first != Ns::No) {
+            forwardTooltipElements = generateTooltipElements(0, forwardNShift.second, true, forwardNShift.first,
+                                                            false,
+                                                            (reverseNShift.second != 0 && reverseNShift.first == Ns::Left));
+        }
+        if (reverseNShift.first != Ns::No) {
+            reverseTooltipElements = generateTooltipElements(0, reverseNShift.second, false, reverseNShift.first,
+                                                            false,
+                                                            (forwardNShift.second != 0 && forwardNShift.first == Ns::Left));
+        }
+    }
+    // generates main part (which contains enzyme)
+    auto generateMainPart = [enzymeSize](const QByteArray& seq, int cut, bool forward) -> QString {
+        QString result;
+        auto append2Result = [&result, forward](const QString& add) {
+            if (forward) {
+                result += add;
+            } else {
+                result.insert(0, add);
+            }
+        };
+        auto removeSpaceFromResult = [&result, forward]() {
+            if (forward) {
+                result = result.left(result.size() - TOOLTIP_SPACE.size());
+            } else {
+                result = result.right(result.size() - TOOLTIP_SPACE.size());
+            }
+        };
+        append2Result(TOOLTIP_SPACE);
+        for (int i = 0; i < enzymeSize; i++) {
+            if (i == cut) {
+                removeSpaceFromResult();
+                append2Result(forward ? TOOLTIP_FORWARD_MARKER : TOOLTIP_REVERSE_MARKER);
+            }
+            append2Result(QString(seq.at(i)));
+            append2Result(TOOLTIP_SPACE);
+        }
+        if (seq.size() == cut) {
+            removeSpaceFromResult();
+            append2Result(forward ? TOOLTIP_FORWARD_MARKER : TOOLTIP_REVERSE_MARKER);
+        }
+        return result;
+    };
+    QString forwardMainPart = generateMainPart(enzyme->seq, enzyme->cutDirect, true);
+    QString reverseMainPart = generateMainPart(seqComplement, enzyme->cutComplement, false);
+
+    // Join parts to a single tooltip
+    auto generateTooltip = [](Ns type, const QStringList& elements, const QString& mainPart) -> QString {
+        QString result;
+        auto appendElements = [&elements]() -> QString {
+            QString appendElementsResult;
+            for (int i = 0; i < elements.size(); i++) {
+                const auto el = elements[i];
```

Comment:
nit: add &, use non-indexed for loop (looks like you do not need `i`

Reply:
OK